libraries/node-core-library/src/Text.ts

```diff
@@ -25,7 +27,12 @@ export const enum NewlineKind {
    * @remarks
    * POSIX is a registered trademark of the Institute of Electrical and Electronic Engineers, Inc.
    */
-  Lf = '\n'
+  Lf = '\n',
+
+  /**
+   * Default newline type for this operating system (`os.EOL`).
+   */
+  OsDefault = 'os'
```

Comment:
FWIW Rush Stack would never recommend this option, because it produces different output on different operating systems, which essentially doubles your test matrix.  As much as possible, we aim for deterministic output on every machine.  This is why our `.gitattributes` file overrides Git to force a specific newline.  I've even considered adding an ESLint rule to discourage usage of `os.EOL`. 😊

I'm fine with providing `OsDefault ` as an option, though.  Determinism may be unimportant to some users.